libraries/freertos_plus/standard/freertos_plus_cli/uart/FreeRTOS_CLI_UART.c

```diff
@@ -0,0 +1,65 @@
+/* FreeRTOS includes. */
+#include "FreeRTOS.h"
+#include "task.h"
+
+#include "FreeRTOS_CLI_UART.h"
+#include "iot_uart.h"
+
+extern IotUARTHandle_t xConsoleUart;
+#define cmdTIME_OUT_MS    ( 500 )
+
+int32_t uart_read( char * const pcInputBuffer,
+                   uint32_t xInputBufferLen );
+void uart_write( const char * const pcOutputBuffer,
+                 uint32_t xOutputBufferLen );
+
+xConsoleIO_t uartConsoleIO =
+{
+    .read  = uart_read,
+    .write = uart_write
+};
+
+int32_t uart_read( char * const pcInputBuffer,
+                   uint32_t xInputBufferLen )
+{
+    int32_t status;
+    uint16_t readSize = 0;
+
+    /* Read to input buffer. */
+    status = iot_uart_read_async( xConsoleUart, ( uint8_t * ) pcInputBuffer, xInputBufferLen );
```

Comment:
Will the layer above this protect concurrency? common-io is not thread safe

Reply:
Yes, it will be used by the console loop in a for loop without any concurrency. I have added also a documentation in `xConsoleIO_t`, if user wishes to use it without Console Loop.